Update Webots instructions for new RoboCup set up (#273)
NUbots/NUbook: src/book/03-guides/02-tools/02-webots.mdx

## Install Webots

[Install Webots 2022b from the GitHub release](https://github.com/cyberbotics/webots/releases/R2022b).

## Running the Official RoboCup Simulation in Webots

### Get the Repository

1. Move into the directory you would like the clone into. The following command will put you in your `home` directory.

```bash
cd ~
```

2. Clone the RoboCup TC Virtual Season repository:

```bash

> git clone https://github.com/RoboCup-Humanoid-TC/hlvs_webots

Reviewer: This is not really necessary.

> git clone https://github.com/RoboCup-Humanoid-TC/hlvs_webots

Reviewer 2: Perhaps not but it's the standard the rest of our clone links follow in nubook

```

### Set Up the Controllers

1. Install dependencies for the referee:

   ```bash
   cd hlvs_webots/controllers/referee
   pip3 install -r requirements.txt
   ```

2. Build the Robocup controllers:

   ```bash
   cd ../..

>    make -j$(nproc)

Reviewer 2: This line fails if you don't have `WEBOTS_HOME` set

## Install Webots

1. [Install Webots 2022b from the GitHub release](https://github.com/cyberbotics/webots/releases/R2022b).

2. Set your `WEBOTS_HOME` environment variable in `~/.bashrc`. Check that `webots` exists in `/usr/local/webots`

   1. Run `nano ~/.bashrc` to view and edit your `bashrc` file.
   2. Add in `export WEBOTS_HOME=/usr/local/webots` (or replace any existing lines with this new path).

   Note that if webots is somewhere else, find it and use that path instead.

   Alternatively you can run `echo export WEBOTS_HOME=/usr/local/webots >> ~/.bashrc` to append the line to the `bashrc` file.

## Running the Official RoboCup Simulation in Webots

### Get the Repository

1. Move into the directory you would like the clone into. The following command will put you in your `home` directory.

```bash
cd ~
```

2. Clone the RoboCup TC Virtual Season repository:

```bash
git clone https://github.com/RoboCup-Humanoid-TC/hlvs_webots.git
```

### Set Up the Controllers

1. Install dependencies for the referee:

   ```bash
   cd hlvs_webots
   pip3 install -r controllers/referee/requirements.txt
   ```

2. Build the Robocup controllers:

   ```bash
   make -j$(nproc)
   ```

>    [Note: you must have `WEBOTS_HOME` set or this will fail](guides/tools/webots-setup#install-webots).

Reviewer 2: Can this be an alert? Also the link is broken because we're already in guides/tools as a subdirectory. Maybe something like this?
```suggestion
   <Alert>
   Note: [You must have `WEBOTS_HOME` set or this will fail](guides/tools/webots-setup#install-webots).
   </Alert>
```

## Install Webots

1. [Install Webots 2022b from the GitHub release](https://github.com/cyberbotics/webots/releases/R2022b).

2. Set your `WEBOTS_HOME` environment variable in `~/.bashrc`. Check that `webots` exists in `/usr/local/webots`

   1. Run `nano ~/.bashrc` to view and edit your `bashrc` file.
   2. Add in `export WEBOTS_HOME=/usr/local/webots` (or replace any existing lines with this new path).

   Note that if webots is somewhere else, find it and use that path instead.

   Alternatively you can run `echo export WEBOTS_HOME=/usr/local/webots >> ~/.bashrc` to append the line to the `bashrc` file.

## Running the Official RoboCup Simulation in Webots

### Get the Repository

1. Move into the directory you would like the clone into. The following command will put you in your `home` directory.

```bash
cd ~
```

2. Clone the RoboCup TC Virtual Season repository:

```bash
git clone https://github.com/RoboCup-Humanoid-TC/hlvs_webots.git
```

### Set Up the Controllers

1. Install dependencies for the referee:

   ```bash
   cd hlvs_webots
   pip3 install -r controllers/referee/requirements.txt
   ```

2. Build the Robocup controllers:

   ```bash
   make -j$(nproc)
   ```

>    <Alert type="warning">
>    Note: [You must have `WEBOTS_HOME` set or this will fail](webots-setup#install-webots).
>    </Alert>

Author: ```suggestion
   <Alert type="warning">
   
   Note: [You must have `WEBOTS_HOME` set or this will fail](webots-setup#install-webots).
   
   </Alert>
```

> 

Reviewer 2: I can't comment on it but the game controller bash commands need to be updated

Follow these steps to set up a NUbots developed environment rather than the RoboCup environment.

1. Clone the NUWebots repository:

   ```bash
   cd ~
   git clone https://github.com/NUbots/NUWebots.git
   ```

2. Move to the `NUWebots` folder:

   ```bash
   cd NUWebots
   ```

3. Install the python dependencies:

   ```bash
   pip3 install -r requirements.txt
   ```

4. Configure the codebase by running:

   ```bash
   ./b configure
   ```

>    [Note: you must have `WEBOTS_HOME` set or this will fail](guides/tools/webots-setup#install-webots).

Reviewer 2: Alert here too pls

Author: ```suggestion
      <Alert type="warning">
      Note: [You must have `WEBOTS_HOME` set or this will fail](webots-setup#install-webots).
      </Alert>
```

Follow these steps to set up a NUbots developed environment rather than the RoboCup environment.

1. Clone the NUWebots repository:

   ```bash
   cd ~
   git clone https://github.com/NUbots/NUWebots.git
   ```

2. Move to the `NUWebots` folder:

   ```bash
   cd NUWebots
   ```

3. Install the python dependencies:

   ```bash
   pip3 install -r requirements.txt
   ```

4. Configure the codebase by running:

   ```bash
   ./b configure
   ```

>       <Alert type="warning">
>       Note: [You must have `WEBOTS_HOME` set or this will fail](webots-setup#install-webots).
>       </Alert>

Author: ```suggestion
      <Alert type="warning">
      
      Note: [You must have `WEBOTS_HOME` set or this will fail](webots-setup#install-webots).
      
      </Alert>
```

Follow these steps to set up a NUbots developed environment rather than the RoboCup environment.

1. Clone the NUWebots repository:

   ```bash
   cd ~
   git clone https://github.com/NUbots/NUWebots.git
   ```

2. Move to the `NUWebots` folder:

   ```bash
   cd NUWebots
   ```

3. Install the python dependencies:

   ```bash
   pip3 install -r requirements.txt
   ```

4. Configure the codebase by running:

   ```bash
   ./b configure
   ```

      <Alert type="warning">
      
      Note: [You must have `WEBOTS_HOME` set or this will fail](webots-setup#install-webots).
      
      </Alert>

> 

Reviewer 2: This one is still broken I think it needs to have less tabs :(

2. Set up the game configuration.

>    1. Open the [game.json file](https://github.com/RoboCup-Humanoid-TC/hlvs_webots/controllers/referee/game.json).

Reviewer 2: This URL is broken, should be
```https://github.com/RoboCup-Humanoid-TC/hlvs_webots/blob/main/controllers/referee/game.json```
